src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -258,30 +258,39 @@ void BindType(NpgsqlTypeMapping mapping, NpgsqlConnector connector, bool throwOn
             if (!found)
             {
                 var msg = $"A PostgreSQL type with the name {mapping.PgTypeName} was not found in the database";
-                if (throwOnError)
+                if (externalCall)
                     throw new ArgumentException(msg);
                 Log.Debug(msg);
                 return;
             }
             else if (pgType == null)
             {
                 var msg = $"More than one PostgreSQL type was found with the name {mapping.PgTypeName}, please specify a full name including schema";
-                if (throwOnError)
+                if (externalCall)
                     throw new ArgumentException(msg);
                 Log.Debug(msg);
                 return;
             }
             else if (pgType is PostgresDomainType)
             {
                 var msg = "Cannot add a mapping to a PostgreSQL domain type";
-                if (throwOnError)
+                if (externalCall)
                     throw new NotSupportedException(msg);
                 Log.Debug(msg);
                 return;
             }
 
             var handler = mapping.TypeHandlerFactory.Create(pgType, connector.Connection);
             BindType(handler, pgType, mapping.NpgsqlDbType, mapping.DbTypes, mapping.ClrTypes);
+
+            if (externalCall)
+                foreach (var domain in DatabaseInfo.DomainTypes)
```

Comment:
nit: I know we don't _need_ the braces here, but the loop body feels long enough to justify them.